File: src/main/java/io/stouder/adonis/service/AdonisRouteService.java
package io.stouder.adonis.service;

import com.intellij.openapi.project.Project;
import io.stouder.adonis.cli.json.routes.RouteDomain;

import java.util.Map;
import java.util.Optional;
import java.util.function.Consumer;

public interface AdonisRouteService {
    static AdonisRouteService getInstance(Project project) {
        return project.getService(AdonisRouteService.class);
    }

    void fetchRoutes(Consumer<Map<String, Optional<RouteDomain[]>>> callback);
}

File: src/main/java/io/stouder/adonis/service/AdonisRouteService.kt
package io.stouder.adonis.service

import com.intellij.openapi.project.Project
import io.stouder.adonis.cli.json.routes.RouteDomain
import java.util.*

interface AdonisRouteService {
    companion object {
        fun getInstance(project: Project): AdonisRouteService {
            return project.getService(AdonisRouteService::class.java)
        }
    }

    fun fetchRoutes(callback: (Map<String, Optional<Array<RouteDomain>>>) -> Unit)
}